src/test/unit/comparison.test.js

```diff
@@ -159,3 +167,206 @@ describe('mergeProfilesForDiffing function', function() {
     ).toBe(false);
   });
 });
+
+describe('mergeThreads function', function() {
+  it('merges the various tables for 2 threads properly', function() {
+    const { profile } = getProfileFromTextSamples(
+      'A[lib:libA]  B[lib:libA]',
+      'A[lib:libA]  A[lib:libB]  C[lib:libC]'
+    );
+
+    const mergedThread = mergeThreads(profile.threads);
+
+    const mergedLibs = mergedThread.libs;
+    const mergedResources = mergedThread.resourceTable;
+    const mergedFunctions = mergedThread.funcTable;
+    const stringTable = mergedThread.stringTable;
+
+    expect(mergedLibs).toHaveLength(3);
+    expect(mergedResources).toHaveLength(3);
+    expect(mergedFunctions).toHaveLength(4);
+
+    // Now check that all functions are linked to the right resources.
+    // We should have 2 A functions, linked to 2 different resources.
+    // And we should have 1 B function, and 1 C function.
+    const libsForA = [];
+    const resourcesForA = [];
+    for (let funcIndex = 0; funcIndex < mergedFunctions.length; funcIndex++) {
```

Comment:
Optional: I get a bit nervous seeing expects in loop logic. It makes tests pretty verbose and hard to understand what they are asserting. Perhaps it would be better to build some human readable strings and assert those, e.g.:

```js
function getFriendlyFuncLibResources(): string {
  let strings = ''
  for (let funcIndex = 0; funcIndex < mergedFunctions.length; funcIndex++) {
    const funcName = stringTable.getString(mergedFunctions.name[funcIndex]);
    const resourceIndex = mergedFunctions.resource[funcIndex];

    let resourceName = '';
    let libName = '';
    if (resourceIndex >= 0) {
      const nameIndex = mergedResources.name[resourceIndex];
      if (nameIndex >= 0) {
        resourceName = stringTable.getString(nameIndex);
      }

      const libIndex = mergedResources.lib[resourceIndex];
      if (libIndex !== null && libIndex !== undefined && libIndex >= 0) {
        libName = mergedLibs[libIndex].name;
      }
    }
    const strings.push(`${funcName} [${resourceName}:${libName}]`)
  }
  return strings;
}

expect(getFriendlyFuncLibResources()).toEqual([
  ...
]);
```

Reply:
Yeah, I think it's better to do it that way. I was mostly copying the test logic we did for other tests again 😄